### PR TITLE
Refactor: boilerplate

### PR DESCRIPTION
**Pull request by kaluabentes**
This PR adds:

- Support to Next.js v8
- Minimalist `next.config.js` configuration
- Minimalist `postcss.config.js` configuration
- Minimalist Storybook configuration
- Support to Stylelint
- Basic folders arquitecture

### CODE REVIEW

File: .storybook/preview-head.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<link href="https://fonts.googleapis.com/css?family=Muli:400,600,700|Rubik:400,500,700|Saira+Extra+Condensed:600" rel="stylesheet" />
```
Comment (caiangums) [inline .storybook/preview-head.html:1]: Why do you need to include fonts?

File: README.md [hunk 2]
```diff
@@ -31,94 +31,3 @@ Now, we are ready to serve our dev server, to do this, please run this command:
...
-| `userMedia` | Object |undefined | User media properties |
-
-
-**Note:** if you don't have yarn installed go to this [link](https://yarnpkg.com/lang/en/docs/getting-started/) and follow the instructions.
-
-Congratulations, your server is running on [http://localhost:5000](http://localhost:5000)!!!
-
-The **first** step of our work is done.
-
-### Second step - Configure the git
```
Comment (caiangums) [inline README.md:102]: Why remove these lines? And if you remove them, Why maintain the `First Step`?

File: package.json [hunk 3]
```diff
@@ -1,64 +1,79 @@
...
     "prop-types": "^15.6.2",
-    "react": "^16.4.2",
-    "react-dom": "^16.4.2",
-    "react-popover": "^0.5.7",
+    "react": "^16.8.3",
+    "react-content-loader": "^3.2.0",
+    "react-day-picker": "^7.2.4",
+    "react-dom": "^16.8.3",
+    "react-fetches": "^1.0.3",
     "react-portal": "^4.1.5",
```
Comment (caiangums) [inline package.json:50]: Is `react-portal` required?

File: package.json [hunk 4]
```diff
@@ -1,64 +1,79 @@
...
-    "fetches": "^0.2.2",
-    "glob": "^7.1.2",
-    "next": "^6.1.1",
-    "next-compose-plugins": "^2.1.1",
+    "compression": "^1.7.3",
+    "creditcard.js": "^2.1.6",
+    "dotenv": "^6.1.0",
+    "express": "^4.16.4",
+    "fetches": "^1.0.1-1.0",
+    "google-map-react": "^1.1.1",
```
Comment (caiangums) [inline package.json:34]: Is `google-map-react` required?

File: src/middlewares/verify-token.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import Cookies from 'js-cookie'
```
Comment (caiangums) [inline src/middlewares/verify-token.js:1]: I think this file is not required. Can you explain why use `Cookies`?

File: src/styles/variables.css [hunk 6]
```diff
@@ -0,0 +1,21 @@
+:root {
+  --color-heliow: #febb01;
```
Comment (caiangums) [inline src/styles/variables.css:2]: There's a lot of color around here. Is this minimal?

File: README.md (hunk 2 above)
Comment (helderberto) [reply README.md:102]: Will upload feature be removed? All these lines are about the upload.

**Comment by caiangums** [comment]
In general, I think you should modify the Boilerplate instead of including your project here and trying to match configs. You forgot to remove files, dependencies, CSS variables and removed a lot of configuration without explaining why you are doing this.

**Review by caiangums — CHANGES_REQUESTED** [review]
For me, it's better if you use the Boilerplate project and change it instead of just trying to merge your project with it.

File: next.config.js [hunk 7]
```diff
@@ -1,71 +1,26 @@
...
 const webpack = require('webpack')
-const glob = require('glob')
 const withPlugins = require('next-compose-plugins')
-const withCSS = require('@zeit/next-css')
-const withSourceMaps = require('@zeit/next-source-maps')
+const css = require('@zeit/next-css')
+const sourceMaps = require('@zeit/next-source-maps')
 const optimizedImages = require('next-optimized-images')
-const withOffline = require('next-offline')
+const fonts = require('next-fonts')
```
Comment (helderberto) [inline next.config.js:6]: `next-fonts` is really necessary for init project?

File: README.md (hunk 2 above)
Comment (caiangums) [reply README.md:102]: This is a component that @beatriz1304 implemented, as far as I remember. I don't know if this component was merged or how this kind of stuff is handled. Maybe Bia can give us a hint on this 😄

File: package.json [hunk 8]
```diff
@@ -1,64 +1,79 @@
...
+    "stylelint": "stylelint ./src/**/*.css",
+    "lint": "yarn eslint && yarn stylelint",
+    "build-storybook": "build-storybook"
   },
-  "author": "Daniel Leite de Oliveira <dleitee@gmail.com> (https://github.com/dleitee)",
+  "husky": {
+    "hooks": {
+      "pre-push": "yarn lint"
+    }
+  },
```
Comment (helderberto) [inline package.json:21]: I think you can move `husky` actions to `.huskyrc` file in root of the project.

File: package.json [hunk 9]
```diff
@@ -1,64 +1,79 @@
...
+  "keywords": [],
+  "author": "",
   "license": "ISC",
   "dependencies": {
-    "@zeit/next-css": "^0.2.0",
+    "@zeit/next-css": "^1.0.2-canary.0",
```
Comment (helderberto) [inline package.json:26]: Isn't a good practice to use versions `beta`, `alpha`, always try the version `stable` of the packages releases.

**Review by helderberto — CHANGES_REQUESTED** [review]
I think project need more descriptions about why make some changes.

File: .storybook/preview-head.html (hunk 1 above)
Comment (kaluabentes) [reply .storybook/preview-head.html:1]: Its for testing purpose

File: server.js [hunk 10]
```diff
@@ -1,34 +1,35 @@
...
-  server.use(handler)
-
-  server.listen(PORT, err => {
-    if (err) throw err
-    console.info(`> Ready on http://localhost:${PORT}`)
+const compression = require('compression')
+const path = require('path')
+
+const isDev = process.env.NODE_ENV !== 'production'
+const port = 3000
```
Comment (natamvo) [inline server.js:7]: poderia ter um process.env.PORT || 3000 e inserir um const URL também com process.env.URL || "http://localhost"

File: README.md (hunk 2 above)
Comment (kaluabentes) [reply README.md:102]: I has moved the upload component to another repo, I've created another repo for the components and removed from here

File: package.json (hunk 3 above)
Comment (kaluabentes) [reply package.json:50]: I will remove the components that its specific to a project

File: package.json (hunk 4 above)
Comment (kaluabentes) [reply package.json:34]: As above

File: src/middlewares/verify-token.js (hunk 5 above)
Comment (kaluabentes) [reply src/middlewares/verify-token.js:1]: For auth token storage

File: src/styles/variables.css (hunk 6 above)
Comment (kaluabentes) [reply src/styles/variables.css:2]: We can remove to

File: next.config.js (hunk 7 above)
Comment (kaluabentes) [reply next.config.js:6]: When we need to work with icon fonts its useful.

File: package.json (hunk 8 above)
Comment (kaluabentes) [reply package.json:21]: I'll do this

**Comment by kaluabentes** [comment]
> For me, it's better if you use the Boilerplate project and change it instead of just trying to merge your project with it.

**Pull request closed without merging**